src/algorithms/clustering/k_means.py

```diff
@@ -1,23 +1,25 @@
 import pandas as pd
 import numpy as np
-from typing import List, Tuple, Generator, Union
+from typing import List, Tuple, Union, Optional
 
-num_types = [int, float, np.float, np.int]
+initial_types = ['random', 'kmeans++']
```

Comment:
makes it sound as though the types are initial. maybe switch to initialization_types, initialize_types or init_types